.github/workflows/discord-notification.yml

```
@@ -0,0 +1,107 @@
+name: Discord Notification
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, closed]
+  push:
+    branches: [main, develop] # Add your target branches
+  issues:
+    types: [opened, closed, reopened]
+  workflow_run:
+    workflows: ["CI", "Tests"] # Add your other workflow names
+    types: [completed]
+
+jobs:
+  notify:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get latest commit message
+        id: commit
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const { data: commits } = await github.rest.pulls.listCommits({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.payload.pull_request.number
+            });
+            const lastCommit = commits[commits.length - 1];
+            core.setOutput('message', lastCommit.commit.message);
+
+      - name: Discord PR Notification
+        if: github.event_name == 'pull_request'
```

```suggestion
        if: github.event_name == 'pull_request' && !github.event.pull_request.draft
```
 avoids unnecessary noise. 👍